scipy/stats/_binned_statistic.py

```diff
@@ -690,15 +688,12 @@ def _bin_numbers(sample, nbin, edges, dedges):
     # Using `digitize`, values that fall on an edge are put in the right bin.
     # For the rightmost bin, we want values equal to the right
     # edge to be counted in the last bin, and not as an outlier.
-    exceptions = (RuntimeWarning,)
-    if sys.version_info >= (3, 8):
-        exceptions += (OverflowError,)  # Python3.8 int(np.inf) throws this
     for i in xrange(Ndim):
         # Find the rounding precision
-        try:
-            decimal = int(-np.log10(dedges[i].min())) + 6
-        except exceptions:
+        dedges_min = dedges[i].min()
+        if dedges_min == 0:
```

Comment:
If `dedges_min` is not an integer this equality will probably never be caught. If it is an integer then the error message is not accurate.

Reply:
But if it is integer valued then it is exactly zero so the error message can omit the `numerically` part.

Reply:
I agree with @ilayn here, numpy has the `np.isclose()` method, would that work with the compared value being `0.0`?

Reply:
I understand you want to deal with precision but I don't think there are many use cases for binned statistics.
I think that in the context of this particular bug we only want `dedges_min` to be numerically different than zero so that np.log does not complain.
And I guess the current implementation achieve this goal.
By the way my test cover the typical mistake where two bins with the same value are provided.

Reply:
If the following is not an issue, which we see once in a while regarding statistics computations, then I am also OK with the current code. I can't judge whether this can happen in this context.

```
>>> np.log(1e-323)
-743.7469247408213

>>> np.log(1e-324)
__main__:1: RuntimeWarning: divide by zero encountered in log
-inf
```

Reply:
Actually 1e-323 != 0 and 1e-324 == 0 on my machine ;-)

Reply:
Yes but we have too many strange architectures that's why I'm splitting hair on this. All of a sudden some build somewhere starts giving errors because zero values are not agreeing on that particular system.

Reply:
I agree with your concern.
To be honest I am in favour not to catch this exception.

Reply:
I'm okay with the exact equality check. @ilayn can you live with it or do you insist on the exception-catching (or some other) route?

Reply:
Let's fix it if fails later. Now I am not able to make a thorough smoke test. So LGTM.